scipy/stats/sampling.py

```diff
@@ -49,3 +70,1262 @@
     SimpleRatioUniforms,
     UNURANError
 )
+
+
+# define pdfs and other helper functions to create the generators
+
+def argus_pdf(x, chi):
```

Comment:
These all look quite simple. Is it an option to write them with Pythran or Cython?

Reply:
that would indeed be a big advantage, it would make the setup much faster. but it requires a change of the interface of NumericalInversePolynomial: it needs to support LowLevelCallable (I think Tirth left a comment about this in the code). or one has to work directly with the UNURAN C code. this could be a nice follow-up project